app/controllers/tasks.js

```diff
@@ -262,16 +276,62 @@ export default class TasksController extends Controller {
       );
     }
   }
-
-  @action async handleUpdateTask(taskId, error) {
+  showTaskChangeInfoModal(msg) {
+    this.message = msg;
+    this.showModal = true;
+    this.buttonRequired = true;
+  }
+  @action async handleUpdateTask(taskId, resetCurrentTask) {
+    this.resetCurrentTask = resetCurrentTask;
     const taskData = this.taskFields;
-    if (taskData.percentCompleted === TASK_PERCENTAGE.completedPercentage) {
-      this.message = TASK_MESSAGES.MARK_DONE;
-      this.showModal = true;
-      this.buttonRequired = true;
-      this.tempTaskId = taskId;
+    const currentTask = this.getTaskById(taskId);
+    const isCurrentTaskStatusBlock =
+      currentTask.status === this.TASK_KEYS.BLOCKED;
+    const isCurrentTaskStatusInProgress =
+      this.TASK_KEYS.IN_PROGRESS === currentTask.status;
+    const isNewTaskStatusInProgress =
+      taskData.status === this.TASK_KEYS.IN_PROGRESS;
+    const isNewTaskStatusBlock = taskData.status === this.TASK_KEYS.BLOCKED;
+    const isCurrProgress100 =
+      parseInt(currentTask.percentCompleted || 0) === 100;
+    const isCurrProgress0 = parseInt(currentTask.percentCompleted || 0) === 0;
+    this.tempTaskId = taskId;
+
+    if (this.dev && taskData.status) {
+      if (
+        (isCurrentTaskStatusBlock || isCurrentTaskStatusInProgress) &&
+        !isCurrProgress100 &&
+        !isNewTaskStatusBlock &&
+        !isNewTaskStatusInProgress
+      ) {
+        this.taskFields.percentCompleted = 100;
+        this.showTaskChangeInfoModal(
+          `The progress of current task is ${currentTask.percentCompleted}%. ${TASK_MESSAGES.CHANGE_TO_100_PROGRESS}`
```

Comment:
The logic for showing the modal dialogue could be consolidated into a separate function to avoid repetition and improve code organization.